### PR TITLE
Add email and calendar reminders to Strategies

### PR DESCRIPTION
**Pull request by andela-oesho**
create self_care_strategy_reminder 
create strategy_reminder 
create relationship between user and strategy
add fields for self care and strategy reminder in strategy view
add rake task to perform strategy reminder
write test for active reminders
write test for notification mailer
refactor notification mailer

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -20,6 +20,16 @@ engines:
     enabled: true
   rubocop:
     enabled: true
+    exclude_fingerprints:
```
Comment (julianguyen) [inline .codeclimate.yml]: What exactly does this do?

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -609,9 +611,14 @@ label {
...
 .label {
   width: auto;
-  display: inherit;
+  display: initial;
   font-size: $font30;
   margin-bottom: 20px;
+  padding: 0;
+  font-weight: bold;
+  line-height: 1;
+  color: inherit ;
```
Comment (julianguyen) [inline app/assets/stylesheets/application.scss]: Can get rid of extra space before semicolon

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -609,9 +611,14 @@ label {
 
 .label {
   width: auto;
-  display: inherit;
+  display: initial;
```
Comment (julianguyen) [inline app/assets/stylesheets/application.scss]: Why is this changed to initial?

File: app/views/notification_mailer/perform_strategy.html.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<p>
+  Hi <%= @user.name %>,
+</p>
+<p>
+  Here's a friendly reminder to perform your strategies: <%= @model.name %>!
```
Comment (julianguyen) [inline app/views/notification_mailer/perform_strategy.html.erb]: This text should be moved into en.yml

File: app/views/notification_mailer/perform_strategy.text.erb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<p>
+  Hi <%= @user.name %>,
+</p>
+<p>
+  Here's a friendly reminder to perform your strategies: <%= @model.name %>!
```
Comment (julianguyen) [inline app/views/notification_mailer/perform_strategy.text.erb]: This text should be moved into en.yml

File: app/views/notification_mailer/refill_medication.html.erb [hunk 6]
```diff
@@ -2,6 +2,6 @@
   Hi <%= @user.name %>,
 </p>
 <p>
-Here's a friendly reminder to refill your medication <%= @medication.name %> on 
-<%= @medication.refill %>!
+Here's a friendly reminder to refill your medication <%= @model.name %> on
```
Comment (julianguyen) [inline app/views/notification_mailer/refill_medication.html.erb]: This text should be moved into en.yml

File: app/views/notification_mailer/refill_medication.text.erb [hunk 7]
```diff
@@ -1,3 +1,3 @@
 Hi <%= @user.name %>,
-Here's a friendly reminder to refill your medication <%= @medication.name %> on 
-<%= @medication.refill %>!
+Here's a friendly reminder to refill your medication <%= @model.name %> on
```
Comment (julianguyen) [inline app/views/notification_mailer/refill_medication.text.erb]: This text should be moved into en.yml

File: app/views/notification_mailer/self_care_strategy.html.erb [hunk 8]
```diff
@@ -0,0 +1,3 @@
+Hi <%= @user.name %>,
+Here's a friendly reminder for your self care strategies <%= @model.name %> on
```
Comment (julianguyen) [inline app/views/notification_mailer/self_care_strategy.html.erb]: This text should be moved into en.yml

File: app/views/notification_mailer/self_care_strategy.text.erb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+Hi <%= @user.name %>,
+Here's a friendly reminder for your self care strategies <%= @model.name %> on
```
Comment (julianguyen) [inline app/views/notification_mailer/self_care_strategy.text.erb]: This text should be moved into en.yml

File: app/views/notification_mailer/take_medication.html.erb [hunk 10]
```diff
@@ -2,5 +2,5 @@
   Hi <%= @user.name %>,
 </p>
 <p>
-  Here's a friendly reminder to take your medication: <%= @medication.name %>!
+  Here's a friendly reminder to take your medication: <%= @model.name %>!
```
Comment (julianguyen) [inline app/views/notification_mailer/take_medication.html.erb]: This text should be moved into en.yml

File: app/views/notification_mailer/take_medication.text.erb [hunk 11]
```diff
@@ -1,2 +1,2 @@
 Hi <%= @user.name %>,
-Here's a friendly reminder to take your medication: <%= @medication.name %>!
+Here's a friendly reminder to take your medication: <%= @model.name %>!
```
Comment (julianguyen) [inline app/views/notification_mailer/take_medication.text.erb]: This text should be moved into en.yml

File: config/locales/en.yml [hunk 12]
```diff
@@ -494,6 +496,11 @@ en:
    name_hint: 'Any activity, train of thought, or form or self-care that helps you'
    viewers_hint: 'Allies who can view your strategy and tag their own moments with it'
    comments_hint: 'Only you and viewers can comment'
+   self_care: "SelfCare strategy"
```
Comment (julianguyen) [inline config/locales/en.yml]: SelfCare should be spelled as Self-care

File: config/locales/en.yml [hunk 13]
```diff
@@ -494,6 +496,11 @@ en:
...
+   self_care_hint: 'A reminder to perform your selfcare strategies'
```
Comment (julianguyen) [inline config/locales/en.yml]: selfcare should be spelled as self-care

File: spec/mailers/notification_mailer_spec.rb [hunk 14]
```diff
@@ -4,19 +4,40 @@
   let(:recipient)  { FactoryGirl.create(:user1, email: "some@user.com") }
   let(:medication) { FactoryGirl.create(:medication, :with_daily_reminder, userid: recipient.id) }
   let(:reminder)   { medication.take_medication_reminder }
+   let(:strategy) { FactoryGirl.create(:strategy, :with_daily_reminder, userid: recipient.id) }
```
Comment (julianguyen) [inline spec/mailers/notification_mailer_spec.rb]: Fix indentation here

**Comment by julianguyen** [comment]
Added a bunch of comments!

File: .codeclimate.yml (hunk 1 above)
Comment (andela-oesho) [inline .codeclimate.yml]: ignores method lines,rubocop should still read lines broken down as one line rather than seeing it as multiline

File: app/assets/stylesheets/application.scss (hunk 2 above)
Comment (andela-oesho) [inline app/assets/stylesheets/application.scss]: cool

File: app/assets/stylesheets/application.scss (hunk 3 above)
Comment (andela-oesho) [inline app/assets/stylesheets/application.scss]: there is an initial fixes the unnecessary padding

File: config/locales/en.yml (hunk 12 above)
Comment (andela-oesho) [inline config/locales/en.yml]: noted

File: config/locales/en.yml (hunk 13 above)
Comment (andela-oesho) [inline config/locales/en.yml]: noted

File: db/migrate/20161113145719_create_strategy_reminders.rb [hunk 15]
```diff
@@ -0,0 +1,10 @@
+class CreateStrategyReminders < ActiveRecord::Migration
+  def change
+    create_table :strategy_reminders do |t|
+    	t.integer :strategy_id, null: false
```
Comment (andela-oesho) [inline db/migrate/20161113145719_create_strategy_reminders.rb]: alright

File: app/assets/javascripts/datepicker.js [hunk 16]
```diff
@@ -0,0 +1,3 @@
+$(document).ready(function(){
```
Comment (julianguyen) [inline app/assets/javascripts/datepicker.js]: This should be:
```
var onReadyDatePicker = function() {
  $('.datepicker').datepicker();
};

$(document).on("page:load ready", onReadyDatePicker);
```

File: app/assets/stylesheets/application.scss [hunk 17]
```diff
@@ -609,9 +611,14 @@ label {
...
   font-size: $font30;
   margin-bottom: 20px;
+  padding: 0;
+  font-weight: bold;
```
Comment (julianguyen) [inline app/assets/stylesheets/application.scss]: Can we remove this?

File: app/assets/stylesheets/application.scss [hunk 18]
```diff
@@ -609,9 +611,14 @@ label {
...
+  line-height: 1;
```
Comment (julianguyen) [inline app/assets/stylesheets/application.scss]: Can we also remove this too?

Comment (andela-oesho) [inline app/assets/stylesheets/application.scss]: have u checked it locally?

File: app/assets/stylesheets/application.scss (hunk 2 above)
Comment (andela-oesho) [inline app/assets/stylesheets/application.scss]: color and padding is needed for the text "Strategy reminder" text to appear

File: app/assets/javascripts/datepicker.js [hunk 19]
```diff
@@ -0,0 +1,5 @@
+var onReadyDatePicker = function() {
```
Comment (julianguyen) [inline app/assets/javascripts/datepicker.js]: Let's get rid of this since we already have a datepicker plugin (see new Medication page)

File: app/assets/stylesheets/application.scss (hunk 3 above)
Comment (julianguyen) [inline app/assets/stylesheets/application.scss]: This change broke the label in sign in
screen shot 2016-11-19 at 8 15 31 pm

(The labels should be blocks)

**Comment by julianguyen** [comment]
screen shot 2016-11-19 at 8 17 59 pm

I think the placement of the reminder should be after the "Allow Comments" section. The label wording is a little confusing. What's the difference between the field with the datepicker and the checkbox?

File: app/assets/javascripts/datepicker.js (hunk 19 above)
Comment (andela-oesho) [inline app/assets/javascripts/datepicker.js]: i tried it but it didnt work, i will look into this on thursday

**Comment by andela-oesho** [comment]
Yes  the check box is for email reminder I will rename it to strategy email reminder , ok will also work on the icon being consistent sorry my UI is poor.

**Comment by julianguyen** [comment]
Going to close this pull request. I cleaned it up in #387 :)
